pkg/local_object_storage/engine/delete.go

```diff
@@ -58,102 +55,25 @@ func (e *StorageEngine) delete(prm DeletePrm) (DeleteRes, error) {
 		defer elapsed(e.metrics.AddDeleteDuration)()
 	}
 
-	var locked struct {
-		is  bool
-		err apistatus.ObjectLocked
-	}
-	var splitInfo *objectSDK.SplitInfo
-
-	// Removal of a big object is done in multiple stages:
-	// 1. Remove the parent object. If it is locked or already removed, return immediately.
-	// 2. Otherwise, search for all objects with a particular SplitID and delete them too.
-	e.iterateOverSortedShards(prm.addr, func(_ int, sh hashedShard) (stop bool) {
-		var existsPrm shard.ExistsPrm
-		existsPrm.SetAddress(prm.addr)
-
-		resExists, err := sh.Exists(existsPrm)
-		if err != nil {
-			if shard.IsErrRemoved(err) || shard.IsErrObjectExpired(err) {
-				return true
-			}
-
-			var splitErr *objectSDK.SplitInfoError
-			if !errors.As(err, &splitErr) {
-				if !shard.IsErrNotFound(err) {
```

Comment:
This one differs from the two implementations.

Reply:
yes, that is the key of my suggestion: why do we differ this logic? we do the same things (and one may always have more or less num of bugs) when delete/inhume objects, why some of them should be collected (if an object is a big one) one way while the other should be handled differently? about this PR: i have improved and fixed a little `Inhume` some time ago and could not understand why i had do it again one more time when looking at `Delete`